importers/djornl/parser.py

```diff
@@ -189,6 +237,12 @@ def process_file(self, file, remap_fn, store_fn, err_list, validator=None):
             err_list.append(f"{file['path']}: no header line found")
             return
 
+        missing_headers = self.check_headers(cols, validator)
```

Comment:
This is in the `process_file` function; after ensuring that the file does actually have content, the first valid (non-comment) line of the file is assumed to be the headers, and is checked against the spec to ensure that all required fields are present.